DOC fixes typo and comment in example plot_grid_search_stats.py

olszewskip opened the pull request:
Regarding suggestion after PR #19966 about a need to comment the modified line.

olszewskip committed:
initial commit

olszewskip committed:
Merge pull request #1 from scikit-learn/main

update base 20.06.2021

olszewskip committed:
Merging branch 'main' into issue_#19949 (after updating fork's base).

olszewskip committed:
In corrected_std func Use kr dummy var Hope to please the linter

olszewskip committed:
Add comment in corrected_std

Explain the kr variable

olszewskip committed:
Resolve a merge conflict. Don't understand why it was there

olszewskip commented:
Continuation of Issue #19966
@lorentzenchr @martinagvilas
I'm not exactly sure what I'm doing, apologies if I'm messing anything up git-wise.
Thank You for being patient with me.
That was my best shot at the comment, feel free to take it from here :)

lorentzenchr commented on examples/model_selection/plot_grid_search_stats.py:
```diff
@@ -190,7 +190,7 @@ def corrected_std(differences, n_train, n_test):
         Variance-corrected standard deviation of the set of differences.
     """
-    kr = len(differences)
+    kr = len(differences)  # k times r, r times repeated k-fold crossvalidation
```
```suggestion
    # kr = k times r, r times repeated k-fold crossvalidation
    kr = len(differences)
```

lorentzenchr reviewed (commented):
Could you also correct the Parameter docstring?

- in both `corrected_std` and `compute_corrected_ttest`
  ```
      differences : ndarray of shape (n_samples, 1)
  ```
  to
  ```python
      differences : ndarray of shape (n_samples,)
  ```
- in both `corrected_std`
  ```python
      Returns
      -------
      corrected_std : int
  ```
  to
  ```python
      Returns
      -------
      corrected_std : float
  ```

lorentzenchr commented:
@olszewskip Thanks for this PR. You seem to have made a new branch from your old branch in #19966. For this PR that's totally fine. The usual way is to sync with (upstream) main and then making a new branch out of the main branch.

olszewskip committed:
Update examples/model_selection/plot_grid_search_stats.py

Co-authored-by: Christian Lorentzen <lorentzen.ch@gmail.com>

olszewskip committed:
Correct Parameter docstring

lorentzenchr reviewed (approved):
Thanks for fixing some more stuff:smirk:

lorentzenchr commented:
@ogrisel @martinagvilas ready to merge in my opinion.

martinagvilas commented on examples/model_selection/plot_grid_search_stats.py:
```diff
@@ -187,9 +187,10 @@ def corrected_std(differences, n_train, n_test):
         Variance-corrected standard deviation of the set of differences.
     """
+    # kr = k times r, r times repeated k-fold crossvalidation
     kr = len(differences)
```
Since in the end `kr` ends up being defined as `len(differences)` it might be worth adding as a comment here something like `kr equals the number of times the models were evaluated`

lorentzenchr replied:
@olszewskip Do you want to add that comment?

olszewskip replied:
Thank You for keeping me in the conversation. I took the liberty of modifying the suggested line a tiny bit.

martinagvilas commented:
@lorentzenchr @olszewskip  LGTM

I have a small suggestion but feel free to disregard if you think is too much.

olszewskip committed:
Comment in corrected_std

Add another line of comment in the body of `corrected_std` function, further clarifying meaning behind the `kr` variable.

lorentzenchr commented on examples/model_selection/plot_grid_search_stats.py:
```diff
@@ -187,9 +187,11 @@ def corrected_std(differences, n_train, n_test):
         Variance-corrected standard deviation of the set of differences.
     """
+    # kr = k times r, r times repeated k-fold crossvalidation,
+    # kr equals the number of times the model was fit and evaluated
```
```suggestion
    # kr equals the number of times the model was evaluated
```

lorentzenchr replied:
I think this is clearer.

olszewskip replied:
Ok. Did I mix up my past tenses, is it "fitted" rather than "fit"? :)

lorentzenchr commented:
@olszewskip CI has failing tests. Maybe, you need to merge main.

glemaitre commented:
I wanted to merge `main` but apparently you removed the rights :)

In addition of merging `main`, could you apply the following patch where I spotted a couple of typo:

```diff
commit e4015a130ccd3dafdf27627bdf59fe2f2b3bc91e
Author: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Date:   Wed Jul 21 17:56:00 2021 +0200

    correct some additional typos

diff --git a/examples/model_selection/plot_grid_search_stats.py b/examples/model_selection/plot_grid_search_stats.py
index cb69e9d264..c4d00b8dfa 100644
--- a/examples/model_selection/plot_grid_search_stats.py
+++ b/examples/model_selection/plot_grid_search_stats.py
@@ -190,8 +190,8 @@ def corrected_std(differences, n_train, n_test):
     corrected_std : float
         Variance-corrected standard deviation of the set of differences.
     """
-    # kr = k times r, r times repeated k-fold crossvalidation,
-    # kr equals the number of times the model was fit and evaluated
+    # kr = k times r, r times repeated k-fold cross-validation,
+    # kr equals the number of times the model was evaluated
     kr = len(differences)
     corrected_var = (
         np.var(differences, ddof=1) * (1 / kr + n_test / n_train)
@@ -281,10 +281,10 @@ print(f"Uncorrected t-value: {t_stat_uncorrected:.3f}\n"
 #
 # Bayesian estimation can be carried out in many forms to answer our question,
 # but in this example we will implement the approach suggested by Benavoli and
-# collegues [4]_.
+# colleagues [4]_.
 #
 # One way of defining our posterior using a closed-form expression is to select
-# a prior conjugate to the likelihood function. Benavoli and collegues [4]_
+# a prior conjugate to the likelihood function. Benavoli and colleagues [4]_
 # show that when comparing the performance of two classifiers we can model the
 # prior as a Normal-Gamma distribution (with both mean and variance unknown)
 # conjugate to a normal likelihood, to thus express the posterior as a normal
@@ -308,7 +308,7 @@ print(f"Uncorrected t-value: {t_stat_uncorrected:.3f}\n"
 #
 # Let's compute and plot the posterior:
 
-# intitialize random variable
+# initialize random variable
 t_post = t(
     df, loc=np.mean(differences),
     scale=corrected_std(differences, n_train, n_test)
@@ -329,7 +329,7 @@ plt.show()
 
 # %%
 # We can calculate the probability that the first model is better than the
-# second by computing the area under the curve of the posterior distirbution
+# second by computing the area under the curve of the posterior distribution
 # from zero to infinity. And also the reverse: we can calculate the probability
 # that the second model is better than the first by computing the area under
 # the curve from minus infinity to zero.
```

olszewskip committed:
Update examples/model_selection/plot_grid_search_stats.py

Co-authored-by: Christian Lorentzen <lorentzen.ch@gmail.com>

olszewskip committed:
Merge branch 'scikit-learn:main' into issue_#19949

olszewskip committed:
glemaitre_2021-07-21_17:56

olszewskip committed:
Merge branch 'scikit-learn:main' into issue_#19949

olszewskip commented:
@martinagvilas @lorentzenchr @glemaitre Thank You for Your guidance. Is there anything I should/could do within this PR?

glemaitre commented:
Everythin is fine. Thanks @olszewskip
